inference/worker/opeanapi_parser.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+import requests
+import yaml
+from oasst_shared.schemas import inference
+
+
+def fetch_openapi_spec(url):
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise Exception(f"Failed to fetch data from URL: {url}. Status code: {response.status_code}")
+
+    content_type = response.headers.get("Content-Type")
+
+    if "application/json" in content_type or url.endswith(".json"):
+        return json.loads(response.text)
+    elif (
+        "application/yaml" in content_type
+        or "application/x-yaml" in content_type
+        or url.endswith(".yaml")
+        or url.endswith(".yml")
+    ):
+        return yaml.safe_load(response.text)
+    else:
+        raise Exception(f"Unsupported content type: {content_type}. Only JSON and YAML are supported.")
+
+
+def get_plugin_config(url: str) -> inference.PluginConfig | None:
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        plugin_dict = response.json()
```

Comment:
here you could go `plugin_config = inference.PluginConfig.parse_obj(plugin_dict)` or even `inference.PluginConfig.parse_raw` directly from the web response. I think a couple of changes like that would make the `getattr` and `setattr` things in the models unnecesssary